### DISCUSSION TITLE
Adding support for Point Clouds

### DISCUSSION THREAD

**Discussion by digital-idiot** [corteva/rioxarray#390] [Ideas]
`rioxarray` can already represent rasters in a tubular `<2D coordinates>, <values>` format. This is very similar to the common point cloud representations we encounter. It would be great if `rioxarray` can be extended to support point clouds as well.

I find it to be extremely hard to do tile-wise processing of point clouds using available means. Either the entire point cloud needs to be loaded into the memory just to process the spatial filter (example: points inside the extent of a rectangular box) or filter the chunks of `n` points in a loop which is extremely slow, and has to be done for each box. I think `rioxarray` has the potential to address this as it has the advantage of geo-awareness as well as all the tricks of manipulating large array from `xarray`.

**Comment by snowman2** [comment]
@digital-idiot, do you have a specific example?

Possibly related to:
- #209 
- #202
- https://github.com/pydata/xarray/issues/3213

Potentially useful:
- https://pdal.io/

**Reply by digital-idiot** [reply]
I did try with [laspy](https://github.com/laspy/laspy/) but apparently tile-wise lazy loading is not possible. Only option is to loop through chunk of `n` points (See: [issue 156](https://github.com/laspy/laspy/issues/156#event-5102877463)). I also went through the documentation of [pdal](https://pdal.io/python.html) but have not seen any way to lazy load only filtered subset into memory.

What I want is to load subset of point cloud within a given extent and feed them to a network. These tiles should be preferably overlapping as required. In case of image this is very straightforward but as you can see for point cloud is not easy given the overheads. In `xarray` this can partially done by the combination of discretizing coordinates and group by operations. However, overlapping can't be achieved (I don't know how) this way. Some form of spatial-indexing / geo-hashing can be quite handy. 

[This issue](https://github.com/xarray-contrib/xoak/issues/12) is kind of related in this context.

**Reply by snowman2** [reply]
rioxarray is scoped to geospatial applications supported by [rasterio](https://github.com/mapbox/rasterio/). After thinking on it, I have concluded that point clouds are not a good fit here unfortunately as it uses rasterio/GDAL instead of more point-cloud focused engines.

You could check in with [geoxarray](https://github.com/geoxarray/geoxarray/) to see if it is something that fits there or this may be a good start for a new [xarray extension](http://xarray.pydata.org/en/stable/ecosystem.html) of its own. Hopefully the references provided in this issue will give some background that will be helpful with this endeavor. I wish you the best in your search for a solution in this domain!
